File: backend/app/schemas/purchasing.py
import uuid
from datetime import date, datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict, Field, field_validator
from app.models.purchasing import PurchaseOrderStatus

class SupplierInput(BaseModel):
    name: str=Field(min_length=1,max_length=160); contact_person: str|None=None; phone: str|None=None; email: str|None=None; address: str|None=None; notes: str|None=None
    @field_validator("name")
    @classmethod
    def name_required(cls,v):
        if not v.strip(): raise ValueError("Supplier name is required.")
        return v.strip()
class SupplierUpdate(BaseModel):
    name: str|None=Field(None,min_length=1,max_length=160); contact_person: str|None=None; phone: str|None=None; email: str|None=None; address: str|None=None; notes: str|None=None; is_active: bool|None=None
class SupplierRead(SupplierInput):
    model_config=ConfigDict(from_attributes=True); id: uuid.UUID; is_active: bool; created_at: datetime; updated_at: datetime
class SupplierPage(BaseModel): items:list[SupplierRead]; page:int; page_size:int; total_items:int; total_pages:int

class POItemInput(BaseModel): product_id:uuid.UUID; quantity:int=Field(gt=0); unit_cost:Decimal=Field(ge=0,max_digits=12,decimal_places=2)
class POCreate(BaseModel): supplier_id:uuid.UUID; expected_date:date|None=None; notes:str|None=None; items:list[POItemInput]=Field(min_length=1)
class POUpdate(POCreate): pass
class ReceiveLine(BaseModel): item_id:uuid.UUID; quantity:int=Field(gt=0)
class ReceiveInput(BaseModel): items:list[ReceiveLine]=Field(min_length=1)
class POItemRead(BaseModel):
    model_config=ConfigDict(from_attributes=True); id:uuid.UUID; product_id:uuid.UUID; product_name:str; sku:str; ordered_quantity:int; received_quantity:int; unit_cost:Decimal; line_total:Decimal
class PORead(BaseModel):
    model_config=ConfigDict(from_attributes=True); id:uuid.UUID; po_number:str; supplier_id:uuid.UUID; supplier:SupplierRead; status:PurchaseOrderStatus; order_date:date; expected_date:date|None; notes:str|None; subtotal:Decimal; items:list[POItemRead]; created_at:datetime; updated_at:datetime
class POPage(BaseModel): items:list[PORead]; page:int; page_size:int; total_items:int; total_pages:int

File: frontend/src/types/purchasing.ts
export interface Supplier { id:string; name:string; contact_person:string|null; phone:string|null; email:string|null; address:string|null; notes:string|null; is_active:boolean; created_at:string; updated_at:string }
export interface SupplierInput { name:string; contact_person?:string|null; phone?:string|null; email?:string|null; address?:string|null; notes?:string|null }
export interface SupplierPage { items:Supplier[]; page:number; page_size:number; total_items:number; total_pages:number }
export type POStatus='DRAFT'|'ORDERED'|'PARTIALLY_RECEIVED'|'RECEIVED'|'CANCELLED'
export interface POItem { id:string; product_id:string; product_name:string; sku:string; ordered_quantity:number; received_quantity:number; unit_cost:string; line_total:string }
export interface PurchaseOrder { id:string; po_number:string; supplier_id:string; supplier:Supplier; status:POStatus; order_date:string; expected_date:string|null; notes:string|null; subtotal:string; items:POItem[]; created_at:string; updated_at:string }
export interface POPage { items:PurchaseOrder[]; page:number; page_size:number; total_items:number; total_pages:number }
export interface POInput { supplier_id:string; expected_date:string|null; notes:string|null; items:Array<{product_id:string;quantity:number;unit_cost:string}> }
